hooks/hostname.hook/README.md

```diff
@@ -1,49 +1,45 @@
 ## Sample counting hook
 
+You can use a hook to create more advanced dynamic hostnames than the simple incremented pattern --- `$\{id\}.example.com` --- from the `hostname` property on a policy. This type of hook calculates the correct hostname and returns that hostname as metadata on the task. To do so, it uses a basic counter system that stores how many nodes have bound to a given policy.
```

Comment:
After this paragraph, let's add:
"This hook is intended to be extended for cases where an external system needs to be contacted to determine the correct hostname. In such a scenario, the new value will still be returned as metadata for the node."

Also, I think it should say:
"...returns that hostname as metadata on the node."